### PR TITLE
Enabling AlwaysSample as default sampler

### PR DESCRIPTION
**Pull request by knabben**
Moving the example to a runnable version on the examples folder. This was based on the otel library ones.

Settings the DefaultSampler to AlwaysSampler in the example, if not set the ProbabiliticSample(1e-4) is used.

### CODE REVIEW

File: newrelic/example_test.go [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-	exporter, err := newrelic.NewExporter("My Service", os.Getenv("NEW_RELIC_API_KEY"))
-	if err != nil {
-		log.Fatal(err)
-	}
-	tp, err := trace.NewProvider(trace.WithSyncer(exporter))
-	if err != nil {
-		log.Fatal(err)
-	}
-	global.SetTraceProvider(tp)
-}
```
Comment (purple4reina) [inline newrelic/example_test.go]: We should keep this example because it shows up in godocs.

Comment (knabben) [reply newrelic/example_test.go]: rolling back, should keep the examples folder as well?

**Comment by willnewrelic** [comment]
Hi @knabben

The AlwaysSampler  change looks good.  What's the advantage to having the runnable example in its own module?

**Comment by knabben** [comment]
The library repository has the runnable examples for different services (web, basic, grpc, etc) if this is not required, the example_test.go is enough.

**Comment by knabben** [comment]
Changing only the example and readme.

**Pull request closed without merging**